Data-feature and data-graphic

mandicai opened the pull request:
To parse graphics for metadata, the script in the kit would previously check whether the HTML had the attribute `data-graphic`. To add the metadata to our log sheet, the kit would rely on the `projectType` in `project.config.js` to determine which tab in the sheet to paste the metadata to.

This does not account for graphics embeds we include in feature repos — those graphics would get pasted in the feature tab of our log sheet. I changed the script to check for both the `data-graphic` (indicates whether the template is a graphic that should be parsed) and `data-feature` (indicates whether the template is a feature and should be parsed) attributes. The `update-log-sheet.js` now gets the type of the project from this `data-` attribute, so that we paste the metadata for graphics embeds into the graphic tab in the log sheet and the metadata for the feature into the feature tab.

## [7.2.0]
### Changed
- `templates/__common__/config/tasks/graphics-meta.js` - check for `data-graphic` and `data-feature`, add `checkForAttribute` function to set the type of each template (graphic or feature) in the metadata based on the `data-` attribute
- `templates/__common__/utils/deployment/update-log-sheet.js` - get the type set in `graphics-meta` to determine the tab in the log sheet to paste metadata to, write metadata to the respective tab unless the hostname is `capybara-test`
- `templates/feature/app/index-queso.html` -> `templates/feature/app/index.html`, `templates/feature/app/index.html` -> `templates/feature/app/templates/index-old.html` - make the Queso feature template the default, move the old index.html to the templates folder
- `templates/feature/app/templates/macros/processors-queso.html`, `templates/feature/app/templates/macros/processors.html` - change styling for lists and subheaders
- `templates/feature/app/templates/base.html`, `templates/graphic/app/templates/base.html` - add default `subject-politics` metadata tag
- `templates/feature/app/templates/base-graphic.html`, `templates/graphic/app/templates/base.html` - add default credits metadata
- `templates/graphic/app/templates/graphic.html` - set tags in graphic embed HTML
- `templates/feature/project.config.js`, `templates/graphic/project.config.js` - add comments to clarify deploy bucket
- `templates/graphic/app/index.html` - remove metadata from graphic `index.html` so it's not added to work log

### Added
- `templates/feature/project.config.js` - add `parserOptions` to feature config file (it previously was only in the graphics config file)
- `templates/feature/app/templates/base-graphic.html`, `templates/feature/app/templates/graphic.html` - add templates for graphic embeds in feature

### To test
1) Go to `data-visuals-create` and checkout the branch `feature-tag`.
2) Navigate to whichever folder you generate features in and run `/path/to/data-visuals-create/bin/data-visuals-create feature feature-tag-test`.
3) Navigate to inside your new test feature. Run `npm run data:fetch`. That will get some sample metadata from the [feature ArchieML template](https://docs.google.com/document/d/1B_jhK1r75fMZVQev8BGU60dgjh1ffE0AxNDZz5dl-RQ/edit).
4) Add the file `graphic.html` to the `app/` folder in the feature and paste in the contents from [this graphic template](https://github.com/texastribune/data-visuals-create/blob/9a30b80139aafd091dc08920c8bd3f6580db16e5/templates/graphic/app/templates/graphic.html). This will simulate scenarios when we create a feature and have embeddable graphics in the same repo, like our COVID-19 tracker.
5) Run `npm run build`. You should see a `dist/` folder. Open up the file `manifest.json` in that folder and look at the graphic metadata that was generated. (I liked to right click in VSCode and hit "Format Document" to make it prettier.)
6) Check that metadata was generated for the feature page and the graphic. The `type` property for each in the manifest file should reflect whether it's a feature or graphic.
7) Next, we're going to check if deploying this feature pastes the graphic metadata to the Embedded tab in [our work log ](https://docs.google.com/spreadsheets/d/1hCP5zGx8dNxk59gI9wBSFY2juJVM8OFCDY45VnNb2nI/edit#gid=965603489)and the feature metadata to the Feature tab. Change the bucket in `project.config.js` to `apps.texastribune.org` — we don't paste `capybara-test` links to the work log because we don't want to paste drafts.
8) Run `npm run deploy`. Check that the metadata was pasted correctly into each tab. **Make sure you delete those rows of metadata when you're done!**

### Todos before merging:
- [x] Update the CHANGELOG

ashley-hebler commented on templates/__common__/config/tasks/graphics-meta.js:
```diff
@@ -159,6 +176,15 @@ const parseGraphic = async (
+  if (await checkForAttribute({ key: 'graphic', page })) {
+    type = 'graphic';
+  }
+
```
if we wanted assume it's either graphic or feature since it got to this point, we could streamline this a bit maybe:
```js
// assume the type is graphic unless data-feature is present
const type = (await page.$('[data-feature]')) ? 'feature' : 'graphic';
  ```

ashley-hebler commented on CHANGELOG.md:
```diff
@@ -10,6 +10,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+### Changed
+- `templates/__common__/config/tasks/graphics-meta.js` - check for `data-graphic` and `data-feature`, set the type of each template (graphic or feature) in the metadata
+- `templates/__common__/utils/deployment/update-log-sheet.js` - get the type set in `graphics-meta` to determine the tab in the log sheet to paste metadata to, write metadata to the respective tab unless the hostname is `capybara-test`
+
```
maybe this comes later in the process, but I noticed there are some other files changed for 🧀 queso classes 🧀

ashley-hebler reviewed (commented):
@mandicai, this seems like a great solve for getting graphics from features and feature project info in the logs sheet. The test steps worked for me, but I ran into an error on deploy, which I'm 95% sure is from my own config issues.

Ultimately, what I started to wonder after seeing how this all works is, do we want to grab the same information from a feature page or `[data-feature]` instance that we do for each graphic project or `data-graphic` instance? I don't know the answer, but a few items come to mind:

**The preview images** - These might be useful still for features, but they kind of end up being a very tall screenshot of the whole page and a pretty large file to have to store in s3. I noticed this [test one](https://s3.amazonaws.com/apps.texastribune.org/features/2021/test-two/preview-large.png) is 2mb. If we don't want them, we'll need to adjust this [not great error catch of mine](https://github.com/texastribune/data-visuals-create/blob/master/templates/__common__/config/tasks/graphics-meta.js#L66) and update the selector to only target `[data-graphic]` selectors instead of `body` or something.

**links** - This is another one that I'm not sure is as useful for features. They're in graphics for cases where we display the graphic as a static image, but it includes links we want to surface. I think an easy way to leave those as empty arrays for features is to more specifically look for `[data-graphic] a` elements [here](https://github.com/texastribune/data-visuals-create/blob/dfe68a2dd9b36a649a6e0c3aedb015b8f6687bcb/templates/__common__/config/tasks/graphics-meta.js#L177)

**tags** - Tags on the other hand do feel somewhat relevant to features. I noticed that the graphic within this feature I tested doesn't have any tags since it's connected to the [feature ArchieML template](https://docs.google.com/document/d/1B_jhK1r75fMZVQev8BGU60dgjh1ffE0AxNDZz5dl-RQ/edit), but wasn't sure if that was just an outcome of the way we're testing. When y'all include embeddable graphics in features IRL, do you usually extend the [graphics base template](https://github.com/texastribune/data-visuals-create/blob/master/templates/graphic/app/templates/base.html)?

mandicai commented:
@ashley-hebler Those are great points. I've added the tags to our feature Google Doc template and made some fixes so that features get tags, that was my mistake for missing that before! I've also modified `graphics-metadata.js` so that it only generates a subset of the metadata attributes for features:

```
[{
  "type": "feature",
  "title": "Phasellus venenatis dapibus ante, vel sodales sem blandit sed.",
  "bucket": "capybara-test.texastribune.org",
  "projectPath": "features/2021/metadata-two",
  "projectURL": "https://capybara-test.texastribune.org/features/2021/metadata-two/",
  "createMonth": "12",
  "createYear": "2021",
  "credits": [" Writer McWriterson", "Author McAuthorson", "Texter McTexterson"],
  "folder": "features/2021/metadata-two",
  "id": "5LkgvS1HAwb1LODNMO0tK",
  "lastBuildTime": "2021-12-15T22:07:58.915Z",
  "label": "index.html",
  "tags": ["subject-politics"]
}]
```
